lib/compile.js

```diff
@@ -105,6 +106,23 @@ function compileLoad(loader, load, compileOpts, cache) {
     if (format == 'cjs')
       mappedLoad.source = mappedLoad.source.replace(hashBangRegEx, '');
     return Promise.resolve()
+    .then(function () {
+      var sourceMap = sourceMappingURL.getFrom(mappedLoad.source);
```

Comment:
If the `metadata.sourceMap` is already set, we should probably skip all of this checking? Or do we still want to support that case as well?